src/main/java/randoop/condition/NameReplacementMap.java

```diff
@@ -0,0 +1,61 @@
+package randoop.condition;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+/** A map from a {@code String} name to another name. */
+public class NameReplacementMap {
+
+  /** The underlying {@code Map} */
+  private final Map<String, String> replacements;
+
+  /** Creates a new {@link NameReplacementMap} with an empty map. */
+  NameReplacementMap() {
+    this.replacements = new HashMap<>();
+  }
+
+  /**
+   * Adds a mapping from {@code originalName} to {@code replacementName}.
+   *
+   * @param originalName the name to be replaced
+   * @param replacementName the replacing name
+   * @return the previous value associated with {@code originalName}, or null if there was no such
+   *     mapping
+   */
+  public String put(String originalName, String replacementName) {
+    return replacements.put(originalName, replacementName);
+  }
+
+  /**
+   * Replace occurrences of names from this map with corresponding replacements.
+   *
+   * @param text the text to search for occurrences of names in this map
+   * @return the text modified by replacing original names with replacement names
+   */
+  String replaceNames(String text) {
+    // make sure that we are replacing from longer to shorter strings to avoid mangled replacement
```

Comment:
OK, I implemented this, but this solution doesn't work either.

If we have the pattern `\b(a|b)\b` replacing "a" first gives the same bad outcome. The method would have to know to pick "b"=>"c" first. In general, there could be cyclic replacements that could not be avoided.

However, in this application the replacements look like "a"=>"x1", "b"=>"x2". So, it is only a problem if the original identifiers from the condition are of the form "xi", for i some number.